code/modules/mob/living/life.dm

```diff
@@ -309,4 +311,22 @@
 	if(client)
 		var/client/C = client
 		for(var/mob/living/carbon/human/H in view(src, world.view))
-			C.images += H.hud_list[NATIONS_HUD]
\ No newline at end of file
+			C.images += H.hud_list[NATIONS_HUD]
+			
+/mob/living/proc/handle_smells()
+	if(istype(get_turf(src), /turf/simulated))
+		var/turf/T = get_turf(src)
+		if(!T.smell)
+			current_smell = null
```

Comment:
You realize, this will instantly make any mob that moves off of a turf that has a "smell" stop smelling, right?